src/safeEval.js

```diff
@@ -5,22 +5,25 @@ function buildSafeEval(unsafeRec, safeEvalOperation) {
 
   const { defineProperties } = Object;
 
-  // We use the the concise method syntax to create an eval without a
-  // [[Construct]] behavior (such that the invocation "new eval()" throws
-  // TypeError: eval is not a constructor"), but which still accepts a
-  // 'this' binding.
-  const safeEval = {
-    eval() {
-      return callAndWrapError(safeEvalOperation, arguments);
-    }
-  }.eval;
+  // We use the the arrow function syntax to create an eval without
+  // a [[Construct]] internal method (such that the invocation "new eval()"
+  // throws "TypeError: eval is not a constructor"), since the 'this' binding
+  // is ignored. This also prevents rejection by the overly eager
+  // rejectDangerousSources.
+  const safeEval = x =>
+    // https://tc39.es/ecma262/#sec-performeval
+    // 2. If Type(x) is not String, return x.
+    typeof x !== 'string' ? x : callAndWrapError(safeEvalOperation, [x]);
 
   // safeEval's prototype RootRealm's value and instanceof Function
   // is true inside the realm. It doesn't point at the primal realm
   // value, and there is no defense against leaking primal realm
   // intrinsics.
 
   defineProperties(safeEval, {
+    // `eval.length` is already initialized correctly.
+    // Ensure that `eval.name` is also correct:
+    name: { value: 'eval' },
```

Comment:
It might be worth making explicit that this property is non-writable, non-enumerable, and non-configurable.

Reply:
This property has the defaults of `{ writeable: false, enumerable: false, configurable: true }`, as it’s defined by **ECMAScript**:
- [Runtime Semantics: InstantiateArrowFunctionExpression]
- [SetFunctionName \( <var>F</var>, <var>name</var>, \[ <var>prefix</var> \] \)][SetFunctionName]

[Runtime Semantics: InstantiateArrowFunctionExpression]: https://tc39.es/ecma262/multipage/ecmascript-language-functions-and-classes.html#sec-runtime-semantics-instantiatearrowfunctionexpression
[SetFunctionName]: https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-setfunctionname